Add an example of `use-escaping` to I18n documentation (#8185)
flutter/website: src/development/accessibility-and-localization/internationalization.md

`AppLocalizations.of(context)!.pronoun("Male")` will default to the
"other" case, and return "they".

> ### Escaping Syntax

Reviewer: ```suggestion
### Escaping syntax
```

> Sometimes, it is necessary to use tokens such as `{` and `}` as normal characters. To ignore
> such tokens from being parsed, enable the `use-escaping` flag by adding the following to `l10n.yaml`.

Reviewer: ```suggestion
Sometimes, you have to use tokens, such as `{` and `}`, as normal characters. To ignore
such tokens from being parsed, enable the `use-escaping` flag by adding the following to `l10n.yaml`.
```

```yaml
use-escaping: true
```

> Then any string of characters wrapped around a pair of single quotes will be ignored during parsing.
> To use a normal single quote character, use a pair of consecutive single quotes. For example,

Reviewer: ```suggestion
The parser ignores any string of characters wrapped with a pair of single quotes.
To use a normal single quote character, use a pair of consecutive single quotes. For example:

```
